components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3299,7 +3300,50 @@ public void deleteOAuthApplication(String consumerKey) throws APIManagementExcep
         }
     }
 
-	public JSONObject resumeWorkflow(Object[] args) {
+    @Override
+    public Application[] getApplicationsByOwner(String userId) throws APIManagementException {
+        return apiMgtDAO.getApplicationsByOwner(userId);
+    }
+
+    @Override
+    public boolean updateApplicationOwner(String userId, Application application) throws APIManagementException {
+
+        boolean isAppUpdated = false;
+
+        try {
+            String oldUserRoles[];
+            RealmService realmService = ServiceReferenceHolder.getInstance().getRealmService();
+            int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(MultitenantUtils.getTenantDomain(username));
+            UserStoreManager userStoreManager = realmService.getTenantUserRealm(tenantId).getUserStoreManager();
+            String oldUserName = application.getSubscriber().getName();
+            oldUserRoles = userStoreManager.getRoleListOfUser(MultitenantUtils.getTenantAwareUsername(oldUserName));
+
+            for (String role : oldUserRoles) {
+                if (role.contains(application.getName())) {
+                    userStoreManager.updateRoleListOfUser(userId, null, new String[] {role});
+                }
```

Comment:
without calling updateRoleListOfUser for each role that contains the application name, a better approach would be to collect all roles which contain application name and then call updateRoleListOfUser once. WDYT?

Reply:
May be we can break the for loop it it's single role per application

Reply:
updateRoleListOfUser will be called only once or rarely twice since there will be only one role matching a particular app name also if the user is having sandbox app only there will be second role name thats why I added inside the loop.
Anyway I have moved the role updating outside the loop since there is a second chance to update the sandbox role

Reply:
@harsha89 Cannot break the loop since there can be 2 roles if we consider the sandbox app